service_client_initializer.go

```diff
@@ -0,0 +1,221 @@
+
+	result := false
+
+	isConsulUp := checkConsulAvailable()
+	if !isConsulUp {
+		return false
+	}
+
+	// Get a new client
+	var host = svc.c.Registry.Host
+	var port = strconv.Itoa(svc.c.Registry.Port)
+	var consulAddr = buildAddr(host, port)
+	consulConfig := consulapi.DefaultConfig()
+	consulConfig.Address = consulAddr
+	client, err := consulapi.NewClient(consulConfig)
+	if err != nil {
+		svc.lc.Error(err.Error())
+		return false
+	}
+
+	services, _, err := client.Catalog().Service(serviceConsulId, "", nil)
+	if err != nil {
+		svc.lc.Error(err.Error())
+		return false
+	}
+	if len(services) <= 0 {
+		svc.lc.Error(serviceConsulId + " service hasn't started...")
+		return false
+	}
+
+	healthCheck, _, err := client.Health().Checks(serviceConsulId, nil)
+	if err != nil {
+		svc.lc.Error(err.Error())
+		return false
+	}
+	status := healthCheck.AggregatedStatus()
+	if status == "passing" {
+		result = true
+	} else {
+		svc.lc.Error(serviceConsulId + " service hasn't been available...")
+		result = false
+	}
+
+	return result
+}
+
+func checkConsulAvailable() bool {
+	addr := fmt.Sprintf("%v:%v", svc.c.Registry.Host, svc.c.Registry.Port)
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		svc.lc.Error(fmt.Sprintf("Consul cannot be reached, address: %v and error is \"%v\" ", addr, err.Error()))
+		return false
+	}
+	conn.Close()
+	return true
+}
+
+func initializeClients() {
+	// initialize Core Metadata clients
+	metaPort := strconv.Itoa(svc.c.Clients[ClientMetadata].Port)
+	metaHost := svc.c.Clients[ClientMetadata].Host
+	metaAddr := buildAddr(metaHost, metaPort)
+	metaPath := v1Addressable
+	metaURL := metaAddr + metaPath
+
+	params := types.EndpointParams{
+		// TODO: Can't use edgex-go internal constants!
+		//ServiceKey:internal.CoreMetaDataServiceKey,
+		ServiceKey:  svc.c.Clients[ClientMetadata].Name,
+		Path:        metaPath,
+		UseRegistry: svc.useRegistry,
+		Url:         metaURL}
+
+	svc.ac = metadata.NewAddressableClient(params, types.Endpoint{})
+
+	params.Path = v1Device
+	params.Url = metaAddr + params.Path
+	svc.dc = metadata.NewDeviceClient(params, types.Endpoint{})
+
+	params.Path = v1DevService
+	params.Url = metaAddr + params.Path
+	svc.sc = metadata.NewDeviceServiceClient(params, types.Endpoint{})
+
+	params.Path = v1Deviceprofile
+	params.Url = metaAddr + params.Path
+	svc.dpc = metadata.NewDeviceProfileClient(params, types.Endpoint{})
+
+	// initialize Core Data clients
+	dataPort := strconv.Itoa(svc.c.Clients[ClientData].Port)
+	dataHost := svc.c.Clients[ClientData].Host
+	dataAddr := buildAddr(dataHost, dataPort)
+	dataPath := v1Event
+	dataURL := dataAddr + dataPath
+
+	params.ServiceKey = svc.c.Clients[ClientData].Name
+	params.Path = dataPath
+	params.UseRegistry = svc.useRegistry
+	params.Url = dataURL
+
+	svc.ec = coredata.NewEventClient(params, types.Endpoint{})
+
+	params.Path = v1Valuedescriptor
+	params.Url = dataAddr + dataPath
```

Comment:
PR #39 isn't applied, dataPath should be changed by params.Path

Reply:
Yes, you're right. It didn't pass the v1Valuedescriptor value.